app/redux/shared_actions/WorkoutActionCreators.js

```diff
@@ -3,11 +3,61 @@ import { Alert } from 'react-native';
 import * as SetsSelectors from 'app/redux/selectors/SetsSelectors';
 import * as AuthSelectors from 'app/redux/selectors/AuthSelectors';
 import { END_WORKOUT } from 'app/ActionTypes';
+import * as Analytics from 'app/utility/Analytics';
+import * as SetEmptyCheck from 'app/utility/transforms/SetEmptyCheck';
 
 export const endWorkout = () => (dispatch, getState) => {
     var state = getState();
     var isWorkoutEmpty = SetsSelectors.getIsWorkoutEmpty(state)
     var isLoggedIn = AuthSelectors.getIsLoggedIn(state);
+    var screenStatus = state.appState.screenStatus;
+    var timeStartActive = null;
+    var timeEndActive = null;
+    var percentAppActive;
+    var sets = state.sets.workoutData;
+    var num_sets = sets.length;
+    var num_sets_with_fields = 0;
+    var num_reps = 0;
+    var num_removes = 0;
+    var num_restores = 0;
+
+    sets.map((set) => {
+        set.reps.map((rep) => {
+            if (rep.removed) {
+                num_removes++;
+            }
+        });
+
+        num_reps += set.reps.length;
+
+        if (SetEmptyCheck.isEmpty(set)) {
```

Comment:
Should be using hasEmptyData not isEmpty, isEmpty also considers reps

Reply:
Fixed